shuup/core/utils/price_display.py

```diff
@@ -214,10 +222,25 @@ def _get_priceful(request, item, quantity):
     should implement the `Priceful` interface.
 
     :type request: django.http.HttpRequest
+    :param request: used as pricing context
     :type item: shuup.core.taxing.TaxableItem
     :type quantity: numbers.Number
+    :type supplier: shuup.core.models.Supplier
+    :param supplier: used to pass for pricing context
     :rtype: shuup.core.pricing.Priceful|None
     """
+    if supplier:
+        # Passed from template and sometimes chosen by end user,
+        # but most of the time just decided by supplier strategy.
+        setattr(request, 'supplier', supplier)
+
+    if hasattr(item, 'supplier'):
```

Comment:
Yeah maybe. How I see it the Non shouldn't be a problem. In the end it is up to the pricing module to survive situation like None in the end the supplier is optional parameter for the module. Also in case the item would be source line and supplier is None then the issue definitely here is not in these price display utils.